src/compiler/parser.ts

```diff
@@ -463,6 +463,13 @@ namespace ts {
     }
 
     /* @internal */
+    export function parseIsolatedEntityName(content: string) {
+        const result = Parser.parseIsolatedEntityName(content);
+        Parser.fixupParentReferences(result.entityName);

```

Comment:
Parent references aren't used in the emit transforms with respect to synthesized nodes. If this is only used in emit, then fixing up the parent references is unneeded.